processing/src/main/java/org/apache/druid/jackson/JacksonModule.java

```diff
@@ -43,28 +47,37 @@ public void configure(Binder binder)
   }
 
   @Provides @LazySingleton @Json
-  public ObjectMapper jsonMapper()
+  public ObjectMapper jsonMapper(Properties props)
   {
-    return new DefaultObjectMapper();
+    return new DefaultObjectMapper(getServiceName(props));
   }
 
   /**
    * Provides ObjectMapper that suppress serializing properties with null values
    */
   @Provides @LazySingleton @JsonNonNull
-  public ObjectMapper jsonMapperOnlyNonNullValue()
+  public ObjectMapper jsonMapperOnlyNonNullValue(Properties props)
   {
-    return new DefaultObjectMapper().setSerializationInclusion(JsonInclude.Include.NON_NULL);
+    return new DefaultObjectMapper(getServiceName(props)).setSerializationInclusion(JsonInclude.Include.NON_NULL);
   }
 
   @Provides @LazySingleton @Smile
-  public ObjectMapper smileMapper()
+  public ObjectMapper smileMapper(Properties props)
```

Comment:
Should we keep the older zero argument method too for ease of use in tests? It wouldn't be a provider method anymore, just a utility.

Reply:
I think it is alright like this. There is just one test that needs zero-arg constructor.